### ISSUE TITLE
Sorry, new size issue.

### ISSUE THREAD

**Issue by Coachmn** [torakiki/pdfsam#448]
Please see #447 for the back story.

I now have two files I'm trying to merge 2.67GB and 1.21GB. I've attached the error log. It looks to me like the 2.67GB gives an integer overflow to a negative value?

[Error Log.txt](https://github.com/torakiki/pdfsam/files/6183567/Error.Log.txt)

**Comment by torakiki** [comment]
What is the Max memory available reported in the About section?

**Comment by Coachmn** [comment]
Max Memory 3 GB.
I have a number of apps open atm and Task Manager reports available memory 10.5 GB. I have 16 GB installed physical memory.

**Comment by torakiki** [comment]
You can try increasing the memory. In the installation directory there is a text file called pdfsam.l4j.ini, open it with a text editor like Notepad++ and change this line `-Xmx512m` into `-Xmx5g` to give it 5GB and then start the application using the `pdfsam.exe`, in the about panel it should show 5GB now

**Comment by Coachmn** [comment]
Don't have that line in the file, what is the full text of the line?

**Comment by Coachmn** [comment]
Sorry, disregard the last comment, added line -Xmx5g and indeed change it to -Xmx7g but still get an error and much the same error log.

**Comment by torakiki** [comment]
Does the Max Memory change in the About section?

**Comment by Coachmn** [comment]
Yes it reflects the setting of -Xmx?g. Later I rebooted the machine so I had clean defragged memory and got the the same error.

**Comment by Coachmn** [comment]
More info.
I took 10 files, 9 around 650MB each, one 145MB and it merged them into a 5.79GB file no problem. Consequently the error must be generated by the input files not the output.

**Comment by torakiki** [comment]
Are those files available somewhere? I can try to test them here to see if it's a memory issue or something with the files

**Comment by Coachmn** [comment]
They are the same files I started with. The bigger files I was trying to use were a combination of the merged files. ie I had merged the first four and then tried to add the next 3 that had also been merged previously. I'll try and send you the small files but it may take a few links. Note that the material is probably still in copyright.

**Comment by barefootPiLgrim** [comment]
hi
is there any progress with this issue please?
I'm having same trouble with files over 1GB...

**Comment by torakiki** [comment]
Can you share the input files with WeTransfer or similar and the memory value in the About section?

**Comment by barefootPiLgrim** [comment]
If you received files from @Coachmn please use those as I can identify with everything.
Same issues and trying for same solutions, like breaking into smaller files to get it done works...

can't share the files due to sensitive information

memory is default 247MB , but like above even when changed it didn't help.
do you think this will solve it now ?

Was there a solution that worked for @Coachmn ? ( I'm tired working around using smaller files )

thanks for a quick reply

**Comment by Coachmn** [comment]
I'm still working with small files!

**Comment by barefootPiLgrim** [comment]
Hi

It doesn't matter what file I'm using or if I'm using one at all.

In split mode without choosing a file I choose option split after page number.

The number of characters I'm putting in throws the above mentioned error?

What is the limit of characters I can put in this field ?

Is there anything you can do about it ?

Regards
